src/main/java/hudson/plugins/jira/JiraSite.java

```diff
@@ -174,8 +200,21 @@
     private transient JiraSession jiraSession = null;
 
     @DataBoundConstructor
+    public JiraSite(URL url, @CheckForNull URL alternativeUrl, @CheckForNull String credentialsId, boolean supportsWikiStyleComment, boolean recordScmChanges, @CheckForNull String userPattern,
+                    boolean updateJiraIssueForAllStatus, @CheckForNull String groupVisibility, @CheckForNull String roleVisibility, boolean useHTTPAuth) {
+        this(url, alternativeUrl, CredentialsHelper.lookupSystemCredentials(credentialsId, url != null ? url.toExternalForm() : null), supportsWikiStyleComment, recordScmChanges, userPattern,
+                updateJiraIssueForAllStatus, groupVisibility, roleVisibility, useHTTPAuth);
+    }
+
+    @Deprecated
```

Comment:
please specify method to use instead of deprecated one and the reason for deprecation

Reply:
Deprecate the previous constructor but leave it in place for Java-level compatibility. See https://jenkins.io/doc/developer/plugin-development/pipeline-integration/ , section "Constructor vs. setters"